webstruct/webannotator.py

```diff
@@ -264,6 +186,107 @@ def _set_base(tree, baseurl):
     head = _ensure_head(tree)
     head.insert(0, Element("base", href=baseurl))
 
+TagPosition = namedtuple('TagPosition', ['element',
+                                         'tag',
+                                         'position',
+                                         'length',
+                                         'is_tail',
+                                         'dfs_number'])
+
+
+def translate_to_dfs(positions, ordered):
+    for position in positions:
+        number = ordered[(position.element, position.is_tail)]
+        yield TagPosition(element=position.element,
+                          tag=position.tag,
+                          position=position.position,
+                          length=position.length,
+                          is_tail=position.is_tail,
+                          dfs_number=number)
+
+
+def enclose(tasks, entity_colors):
+    if not tasks:
+        return
+
+    first = tasks[0][0]
+    element = first.element
+    is_tail = first.is_tail
+    source = element.text
+    if is_tail:
+        source = element.tail
+
+    if not source or not source.strip():
+        return
+
+    remainder = source[:first.position]
+
+    nodes = list()
+    for number, (start, end, _id) in enumerate(tasks):
```

Comment:
number is a very uncommon name for index variables; would you mind using "idx" or "index"?